dev/ToastNotifications/NotificationProperties.cpp

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "ToastNotification.h"
+#include "NotificationProperties.h"
+#include "ToastNotificationUtility.h"
+#include "../PushNotifications/PushNotificationUtility.h"
+
+namespace winrt
+{
+    using namespace winrt::Microsoft::Windows::ToastNotifications;
+}
+
+namespace ToastABI
+{
+    using namespace ::ABI::Microsoft::Internal::ToastNotifications;
+}
+
+namespace Helpers
+{
+    using namespace winrt::Microsoft::Windows::PushNotifications::Helpers;
+}
+
+NotificationProperties::NotificationProperties(winrt::ToastNotification const& toastNotification)
+{
+    // Extract payload and convert it from XML to a byte array
+    auto payload = toastNotification.Payload();
+    auto payloadAsHstring = payload.GetXml();
+
+    std::wstring payloadAsStdWstring{ payloadAsHstring };
```

Comment:
No need to allocate more memory here. Just remove the local and use wil::str_raw_ptr below